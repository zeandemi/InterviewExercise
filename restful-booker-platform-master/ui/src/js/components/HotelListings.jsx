import React from 'react';
import fetch from 'node-fetch';
import HotelListing from './HotelListing.jsx';
import HotelForm from './HotelForm.jsx';

export default class HotelListings extends React.Component {

    constructor() {
        super();
		this.state = {
						hotels : []
					};
		
		this.updateHotels = this.updateHotels.bind(this);
    }
    
    componentDidMount() {
		fetch('http://' + window.location.hostname + ':3001/hotel')
			.then(res => res.json())
			.then(body => {
				this.setState({hotels : body});
			});
	}

	updateHotels() {
		fetch('http://' + window.location.hostname + ':3001/hotel')
			.then(res => res.json())
			.then(body => {
				this.setState({hotels : body});
			});
	}

    render() {
		let inputFields = null;
		
		if(this.props.isAuthenticated){
			inputFields = <HotelForm updateHotels={this.updateHotels}/>
		}
	  
		return(
			<div>
				<div className="row">
					<div className="col-sm-2 rowHeader"><p>Hotel name</p></div>
					<div className="col-sm-3 rowHeader"><p>Address</p></div>
					<div className="col-sm-2 rowHeader"><p>Owner</p></div>
					<div className="col-sm-2 rowHeader"><p>Phone number</p></div>
					<div className="col-sm-2 rowHeader"><p>Email</p></div>
					<div className="col-sm-1"></div>
				</div>
				{this.state.hotels.map((hotel, index) => {
					return <div key={hotel.hotelid}><HotelListing details={hotel} updateHotels={this.updateHotels} isAuthenticated={this.props.isAuthenticated}/></div>
				})}
				{inputFields}
			</div>
		);
    }
  }