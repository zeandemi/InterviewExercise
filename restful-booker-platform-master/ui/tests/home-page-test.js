import React from 'react';
import HotelListings from '../src/js/components/HotelListings.jsx';

// We first declare our test by using Jests test function and provide it
// with a name and then an anonymous function which will run our test
test('Hotels list component', () => {
    // Our component we want to test requires data so we create
    // our object to pass later
    const hotelPayload = {
        "hotelid": 1,
        "name": "Hilton",
        "address": "123 The Street",
        "regdate": "2018-02-20T22:47:38.176Z",
        "contact": {
            "name": "Mark Winteringham",
            "phone": "07411222473",
            "email": "mark@hilton.com"
        }
    }

    // Next we create our component that we want to check. By using
    // enzymes shallow function and passing it the React component
    // we want to create, Enzyme will create a headless DOM and render
    // the React component inside.
    const hotels = shallow(
        <HotelListings />
    );

    // Now that we have a rendered React component, we can call it's
    // internal functions. We call setState and pass it our data to 
    // update it's internal state. The component will then process that
    // data and do something with it.
    hotels.setState({hotels : [hotelPayload]});

    // We then compare what the component has done to a previously saved
    // state that lives in the __snapshots__ folder. If anything has changed
    // it will raise an alert.
    expect(hotels).toMatchSnapshot();
});

// Check suggestions...
//
// Check creation of components are consistent
// Check state changes in components that use isAuthorised
// Check BookingListings populates with BookingListing components