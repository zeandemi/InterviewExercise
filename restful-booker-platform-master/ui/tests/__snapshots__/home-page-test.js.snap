// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotels list component 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-sm-2 rowHeader"
    >
      <p>
        Hotel name
      </p>
    </div>
    <div
      className="col-sm-3 rowHeader"
    >
      <p>
        Address
      </p>
    </div>
    <div
      className="col-sm-2 rowHeader"
    >
      <p>
        Owner
      </p>
    </div>
    <div
      className="col-sm-2 rowHeader"
    >
      <p>
        Phone number
      </p>
    </div>
    <div
      className="col-sm-2 rowHeader"
    >
      <p>
        Email
      </p>
    </div>
    <div
      className="col-sm-1"
    />
  </div>
  <div
    key="1"
  >
    <HotelListing
      details={
        Object {
          "address": "123 The Street",
          "contact": Object {
            "email": "mark@hilton.com",
            "name": "Mark Winteringham",
            "phone": "07411222473",
          },
          "hotelid": 1,
          "name": "Hilton",
          "regdate": "2018-02-20T22:47:38.176Z",
        }
      }
      updateHotels={[Function]}
    />
  </div>
</div>
`;
